compiler/core/src/bytecode.rs

```diff
@@ -562,7 +578,29 @@ pub enum Instruction {
 }
 const _: () = assert!(mem::size_of::<Instruction>() == 1);
 
-#[derive(Copy, Clone, Serialize, Deserialize)]
+impl From<Instruction> for u8 {
+    #[inline]
+    fn from(ins: Instruction) -> u8 {
+        // SAFETY: there's no padding bits
+        unsafe { std::mem::transmute::<Instruction, u8>(ins) }
+    }
+}
+
+impl TryFrom<u8> for Instruction {
+    type Error = crate::marshal::MarshalError;
+
+    #[inline]
+    fn try_from(value: u8) -> Result<Self, crate::marshal::MarshalError> {
+        if value <= u8::from(Instruction::ExtendedArg) {
+            Ok(unsafe { std::mem::transmute::<u8, Instruction>(value) })
+        } else {
+            Err(crate::marshal::MarshalError::InvalidBytecode)
+        }
+    }
+}
```

Comment:
Yeah, the variants have fields but because they're all ZSTs this is sound. And if Instruction ever got bigger than a u8 then the `mem::transmute()` would start throwing a compile error